src/ui/public/index_patterns/unsupported_time_patterns.js

```diff
@@ -0,0 +1,34 @@
+import { BoundToConfigObjProvider } from 'ui/bound_to_config_obj';
+
+export function UnsupportedTimePatternsProvider(Private, $injector) {
+  const BoundToConfigObj = Private(BoundToConfigObjProvider);
+  const sessionStorage = $injector.get('sessionStorage');
+  const kbnUrl = $injector.get('kbnUrl');
+  const Notifier = $injector.get('Notifier');
+
+  const notify = new Notifier({ location: 'Index Patterns' });
+
+  const HISTORY_STORAGE_KEY = 'indexPatterns:warnAboutUnsupportedTimePatterns:history';
+  const FLAGS = new BoundToConfigObj({
+    enabled: '=indexPatterns:warnAboutUnsupportedTimePatterns'
+  });
+
+  return function warnAboutUnsupportedTimePattern(indexPattern) {
+    if (!FLAGS.enabled) {
+      return;
+    }
+
+    const previousIds = sessionStorage.get(HISTORY_STORAGE_KEY) || [];
+    if (previousIds.includes(indexPattern.id)) {
+      return;
+    }
+
+    sessionStorage.set(HISTORY_STORAGE_KEY, [...previousIds, indexPattern.id]);
+    notify.warning(
```

Comment:
Maybe we just need a different name. What do you think of my idea of making this module's single responsibility to be to check our flag and persistence rules? Then we'd end up with something like this:

```javascript

import { BoundToConfigObjProvider } from 'ui/bound_to_config_obj';

export function IsUserAwareOfUnsupportedTimePatternProvider(Private, $injector) {
  const BoundToConfigObj = Private(BoundToConfigObjProvider);
  const sessionStorage = $injector.get('sessionStorage');

  const HISTORY_STORAGE_KEY = 'indexPatterns:warnAboutUnsupportedTimePatterns:history';
  const FLAGS = new BoundToConfigObj({
    enabled: '=indexPatterns:warnAboutUnsupportedTimePatterns'
  });

  return function isUserAwareOfUnsupportedTimePattern(indexPattern) {
    // The user's disabled the notification. They know about it.
    if (!FLAGS.enabled) {
      return true;
    }

    // We've already told the user.
    const previousIds = sessionStorage.get(HISTORY_STORAGE_KEY) || [];
    if (previousIds.includes(indexPattern.id)) {
      return true;
    }

    // Let's store this for later, so we don't tell the user multiple times.
    sessionStorage.set(HISTORY_STORAGE_KEY, [...previousIds, indexPattern.id]);
    return false;
  };
}
```

And in `index_pattern.js`, we'd use it like this:

```javascript
    if (indexPattern.isUnsupportedTimePattern()) {
      if (!isUserAwareOfUnsupportedTimePattern(indexPattern)) {
        notify.warning(
          'Support for time-intervals has been removed. ' +
          `View the ["${indexPattern.id}" index pattern in management](` +
          kbnUrl.getRouteHref(indexPattern, 'edit') +
          ') for more information.'
        );
      }
    }
```